File: exercism/typescript/sum-of-multiples/sum-of-multiples.ts
export function sum(multiples: number[], limit: number): number {
  let sum: number = 0;
  for (let curr = 0; curr < limit; curr++) {
    for (let multiple of multiples) {
      if (curr % multiple === 0) {
        sum += curr;
        break;
      }
    }
  }

  return sum;
}

File: exercism/python/sum-of-multiples/sum_of_multiples.py
def sum_of_multiples(limit: int, multiples: list[int]) -> int:
    result: set[int] = set()
    for i in range(limit):
        for multiple in multiples:
            if multiple == 0:
                continue
            if i % multiple == 0:
                result.add(i)
    return sum(result)
